indico/legacy/pdfinterface/conference.py

```diff
@@ -366,7 +366,7 @@ def key_func(c):
                     return True, None
                 return False, speakers[0].get_full_name(last_name_upper=False, abbrev_first_name=False).lower()
         else:
-            key_func = attrgetter(mapping.get(sort_by) or 'title')
+            key_func = attrgetter(mapping.get(sort_by) or sort_by or 'title')
```

Comment:
`sort_by` is supposed to be an enum value (from `BOASortField`), and `attrgetter()` won't work with such a value.

`mapping.get(sort_by, sort_by.value) or 'title'` seems like the safer choice to also allow the special case of passing a string there (which is done only when called from `RHContributionsExportPDFBookSorted`). Or maybe we should just add `board_number` as a new option to the enum?

Reply:
Actually, I wonder for which cases we added `mapping = {'number': 'id', 'name': 'title'}`... maybe legacy code back then that was passing strings instead of enum values.

Reply:
Reworked to add more enums. This also adds the possibility to sort by session or schedule and boardnumber  inside the sessions.

And from a fast git grep it really seems like the mapping part is actually unused now.